Sources/MongoSwift/Operations/Operation.swift

```diff
@@ -21,15 +40,36 @@ internal struct DefaultOperationExecutor: OperationExecutor {
     internal func execute<T: Operation>(_ operation: T,
                                         client: MongoClient,
                                         session: ClientSession?) throws -> T.OperationResult {
-        // if a session was provided, use its underlying connection
-        if let session = session {
-            let conn = try session.getConnection(forUseWith: client)
+        switch operation.connectionStrategy {
+        case let .bound(conn):
+            // pass in the connection this operation is already bound to
             return try operation.execute(using: conn, session: session)
+        case .unbound:
```

Comment:
We could consolidate all the checking in / checking out of connections to this spot by adding some more checks. 
- For instance, after executing an `.unbound` operation, we could check to see if it has become `.bound`. If so, then don't check the connection back in. Otherwise, do check it in. This would essentially be the "stealing" behavior that was once here. 
- Similarly, after executing a `.bound` connection, we could check if it has become `.unbound`, and if so, check the connection back in. This case would be if a `NextOperation` failed on a cursor or something like that.

What do you think about this approach? I slightly prefer it since it would restrict a lot of the connection lifetime management to just this method, rather than having calls to `resolveConnection` and `checkIn` in more places. Some cases will need to remain no matter what though, like if a cursor goes out of scope before it's exhausted.

Reply:
consolidating the logic seems nice. I tried to test this out but quickly ran into some problems surrounding mutability - the `Operation.execute` method would need to become mutating for this to work (so we could change the `connectionStrategy` there), and also `OperationExecutor.execute` would need to take the operation as an `inout` parameter which feels kind of weird.
I think, for now, maybe we should just stick with how things are and open a ticket about exploring this more at a future point?

Reply:
For what it's worth, the Rust Operation layer's equivalents are mutating / take in mutable references. That said, I think it's fine if we defer further work on this since we need to get through some other higher priority things pre 1.0. I filed [SWIFT-633](https://jira.mongodb.org/browse/SWIFT-633) for that.

Reply:
👍 ty!